src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -294,6 +293,61 @@ public bool IsSolutionAvailable
             }
         }
 
+        public IEnumerable<string> GetDeferredProjectsFilePath()
```

Comment:
Blocking UI thread isn't a good idea, but here it's worker thread, and I believe it won't result into deadlock. It depends on case to case basis when to use what. And I also beg to differ with your justification for `JTF.Run` it could be used any where you want to execute asynchronous code inside your synchronize API such is the case here since I need to switch to Main thread. Otherwise I wouldn't have used it at all. So I would like to go ahead with this.